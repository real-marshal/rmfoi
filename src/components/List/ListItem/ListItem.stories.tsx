import type { Meta, Story } from '@storybook/react'
import { ListItem, ListItemProps } from './ListItem'

export default {
  title: 'Common/List/ListItem',
  component: ListItem,
} as Meta

const Template: Story<ListItemProps> = (args) => <ListItem {...args} />

export const Basic = Template.bind({})
Basic.args = {
  children: 'List Item',
}

export const Selected = Template.bind({})
Selected.args = {
  children: 'List Item Selected',
  selected: true,
}
