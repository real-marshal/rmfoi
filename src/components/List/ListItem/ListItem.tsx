import { dt } from '@/features/Theming'
import { css } from '@emotion/react'
import styled from '@emotion/styled'
import type { LiHTMLAttributes, ReactNode } from 'react'

interface ListItemProps extends LiHTMLAttributes<HTMLLIElement> {
  children: ReactNode
  selected?: boolean
}

const StyledLI = styled.li<ListItemProps>`
  color: ${dt.theme.primary};
  background-color: ${dt.theme.background};
  list-style: none;
  padding: ${dt.padding.xl} ${dt.padding.xl2};
  cursor: pointer;

  &:first-of-type {
    border-radius: 6px 6px 0 0;
  }

  &:last-of-type {
    border-radius: 0 0 6px 6px;
  }

  &:hover {
    color: ${dt.theme.background};
    background-color: ${dt.theme.primary};
  }

  ${({ selected }) =>
    selected &&
    css`
      color: ${dt.theme.background};
      background-color: ${dt.theme.primary};
    `}
`

const ListItem = ({ children, ...props }: ListItemProps) => (
  <StyledLI {...props}>{children}</StyledLI>
)

export { ListItem }
export type { ListItemProps }
