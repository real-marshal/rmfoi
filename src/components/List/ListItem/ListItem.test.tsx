import { composeStories } from '@storybook/react'
import { render } from '@testing-library/react'
import * as stories from './ListItem.stories'

describe('ListItem component', () => {
  const testStories = Object.values(composeStories(stories)).map(
    (Story) => [Story.storyName, Story] as const
  )

  it.each(testStories)('renders %s story', (_, Story) => {
    const view = render(<Story />)
    expect(view.baseElement).toMatchSnapshot()
  })
})
