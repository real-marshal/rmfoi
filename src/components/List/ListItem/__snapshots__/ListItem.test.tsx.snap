// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders Basic story 1`] = `
.emotion-0 {
  color: var(--dt-theme-primary);
  background-color: var(--dt-theme-background);
  list-style: none;
  padding: var(--dt-padding-xl) var(--dt-padding-xl2);
  cursor: pointer;
}

.emotion-0:first-of-type {
  border-radius: 6px 6px 0 0;
}

.emotion-0:last-of-type {
  border-radius: 0 0 6px 6px;
}

.emotion-0:hover {
  color: var(--dt-theme-background);
  background-color: var(--dt-theme-primary);
}

<body>
  <div>
    <li
      class="emotion-0 emotion-1"
    >
      List Item
    </li>
  </div>
</body>
`;

exports[`ListItem component renders Selected story 1`] = `
.emotion-0 {
  color: var(--dt-theme-primary);
  background-color: var(--dt-theme-background);
  list-style: none;
  padding: var(--dt-padding-xl) var(--dt-padding-xl2);
  cursor: pointer;
  color: var(--dt-theme-background);
  background-color: var(--dt-theme-primary);
}

.emotion-0:first-of-type {
  border-radius: 6px 6px 0 0;
}

.emotion-0:last-of-type {
  border-radius: 0 0 6px 6px;
}

.emotion-0:hover {
  color: var(--dt-theme-background);
  background-color: var(--dt-theme-primary);
}

<body>
  <div>
    <li
      class="emotion-0 emotion-1"
    >
      List Item Selected
    </li>
  </div>
</body>
`;
